File: web/src/types/video.ts
export type VideoStatus = 'pending' | 'processing' | 'ready' | 'failed'

export interface VideoListItem {
  id: string
  youtube_id: string
  title: string
  channel: string | null
  duration_seconds: number | null
  thumbnail_url: string | null
  status: VideoStatus
  error_message: string | null
  created_at: string
}

export interface Segment {
  id: string
  label: string
  summary: string
  start_time: number
  end_time: number
}

export interface VideoDetail extends VideoListItem {
  transcript_source: string | null
  segments: Segment[]
}

File: api/app/schemas/video.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class VideoCreate(BaseModel):
    url: str


class VideoListItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    youtube_id: str
    title: str
    channel: str | None
    duration_seconds: int | None
    thumbnail_url: str | None
    status: str
    error_message: str | None
    created_at: datetime


class VideoDetail(VideoListItem):
    transcript_source: str | None
